src/config.rs

```diff
@@ -37,6 +42,13 @@ impl GlobalConfig {
         Self {
             level: None,
             handlebars: make_handlebars_registry(),
+            run_id: LazyLock::new(|| {
+                let rng = rand::rng();
+                rng.sample_iter(rand::distr::Alphanumeric)
+                    .take(16)
```

Comment:
16 might be a bit too much. Try generating a bunch of witnesses in a directory from a few different runs and see what the directory feels like. Are you able to find a given crate from a given run easily? Are all the prefixes helpful, or do they cross into being annoyingly long?

`rand::distr::Alphanumeric` has 62 options, which is about 2^6. The probability of collision here follows the [Birthday Paradox setup](https://en.wikipedia.org/wiki/Birthday_problem), and we'd probably be quite happy with somewhere in the region of 1-in-100k / 1-in-1M chance of collision. If my math is correct, ~6 chars should be plenty for that, though it's late and I'm tired so it's probably best if you check my math :)